BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLib.c

```diff
@@ -64,15 +79,28 @@ CpuInit (
   IN UINT32     Index
   )
 {
-  UINT32  ApicId;
+  SMMBASE_INFO  *SmmBaseInfo;
+  UINT32         ApicId;
+  UINT32         CpuIdx;
 
   ApicId = GetApicId();
   if (Index < PcdGet32 (PcdCpuMaxLogicalProcessorNumber)) {
     mSysCpuInfo.CpuInfo[Index].ApicId = ApicId;
   }
 
-  if (PcdGetBool (PcdSmmRebaseEnabled)) {
-    SmmRebase (Index, ApicId);
+  if (PcdGet8 (PcdSmmRebaseEnabled) == SMM_REBASE_ENABLE_S3) {
+    if (GetBootMode() == BOOT_ON_S3_RESUME) {
+      SmmBaseInfo = (SMMBASE_INFO *)PcdGet32 (PcdSmramTsegBase);
+      if (SmmBaseInfo->Signature == PLD_TO_LDR_SMM_SIG) {
+        for (CpuIdx = 0; CpuIdx < SmmBaseInfo->CpuEntry; CpuIdx++) {
+          if (ApicId == SmmBaseInfo->SmmBase[CpuIdx].ApicId) {
+            SmmRebase (Index, ApicId, SmmBaseInfo->SmmBase[CpuIdx].SmmBase - PcdGet32 (PcdSmramTsegBase));
```

Comment:
Add a "break" to optimize the loop flow.
And if we can not find the ApicId in the SMMBASE table,  need to assert().

Reply:
Added the code.